fl33t/models/mixins.py

```diff
@@ -25,20 +27,51 @@ def builds(self, offset=None, limit=None):
 
 class OneBuildMixin:  # pylint: disable=too-few-public-methods
     """For models with a build parent"""
+
+    _build = None
+
+    @property
     def build(self):
         """Return the parent build"""
-        return self._client.get_build(self.build_id)
+        if not self.build_id:
+            return None
+
+        if not self._build or self.build_id != self._build.build_id:
+            if hasattr(self, 'fleet') and self.fleet.train_id:
+                train_id = self.fleet.train_id
+
+            elif hasattr(self, 'train_id') and self.train_id:
+                train_id = self.train_id
+
+            else:
+                return None
+
+            self._build = self._client.get_build(train_id, self.build_id)
+
+        return self._build
 
 
 class OneTrainMixin:  # pylint: disable=too-few-public-methods
     """For models with a train parent"""
+
+    _train = None
+
+    @property
     def train(self):
         """Return the parent train"""
-        return self._client.get_train(self.train_id)
+
+        if not self.train_id:
```

Comment:
I feel like, for the sake of maintaining consistency, we should always check `self._train.train_id` instead of this pseudo-double-cache here.

Reply:
Waiting on the agreed solution regarding the `build` property so that it can be applied here too.